configfile.js

```diff
@@ -338,17 +338,29 @@ cfreader.load_config = function (name, type, options) {
 
     if (!fs.existsSync(name)) {
 
-        if (!/\.json$/.test(name)) {
+        if (!/\.h?json$/.test(name)) {
             return cfrType.empty(options, type);
         }
 
-        const yaml_name = name.replace(/\.json$/, '.yaml');
-        if (!fs.existsSync(yaml_name)) {
-            return cfrType.empty(options, type);
+        if (/\.hjson$/.test(name)) {
+            const json_name = name.replace(/\.hjson$/, '.json');
+            if (!fs.existsSync(json_name)) {
+                return cfrType.empty(options, type);
+            }
+
+            name = json_name;
+            type = 'json';
+        }
+        else {
+            const yaml_name = name.replace(/\.json$/, '.yaml');
+            if (!fs.existsSync(yaml_name)) {
+                return cfrType.empty(options, type);
+            }
+
+            name = yaml_name;
+            type = 'yaml';
         }
```

Comment:
This entire section doesn't seem right. The gist of this section is to let a .yaml file override a missing json file. You want to extend it to let a yaml override a hjson or json file. The easiest way to fix it is to revert this section back and then make this alteration:

```diff
index 05836a7..e77c0a6 100644
--- a/configfile.js
+++ b/configfile.js
@@ -338,7 +338,7 @@ cfreader.load_config = function (name, type, options) {
 
     if (!fs.existsSync(name)) {
 
-        if (!/\.json$/.test(name)) {
+        if (!/\.h?json$/.test(name)) {
             return cfrType.empty(options, type);
         }
 

```

Reply:
I made it so HJSON can be overriedn by JSON and also YAML, but I might be worng

Reply:
Hrmm, I'm wondering how useful that is. Using YAML in place of JSON?  Easy to see the use case. Using HJSON instead of JSON, against easy to see. Using JSON in place of HJSON? Why not just specify JSON?  What's the point?  (If there is a good point, it should be added in the docs.)

Reply:
Ok, I get it. So, should I change it to allow both to be overridden by YAML? Am I right?

Reply:
Correct. Basically treat hjson the same as json.